planning/route_handler/include/route_handler/route_handler.hpp

```diff
@@ -246,6 +248,22 @@ class RouteHandler
   bool getClosestLaneletWithConstrainsWithinRoute(
     const Pose & search_pose, lanelet::ConstLanelet * closest_lanelet, const double dist_threshold,
     const double yaw_threshold) const;
+
+  /**
+   * Finds the closest route lanelet to the search pose satisfying the distance and yaw constraints
+   * with respect to a reference lanelet, the search set will include previous route lanelets,
+   * next route lanelets, and neighbors of reference lanelet. Returns false if failed to find
+   * lanelet.
+   * @param search_pose pose to find closest lanelet to
+   * @param reference_lanelet reference lanelet to decide the search set
+   * @param dist_threshold distance constraint closest lanelet must be within
+   * @param yaw_threshold yaw constraint closest lanelet direction must be within
+   */
+  bool getClosestRouteLaneletFromLanelet(
+    const Pose & search_pose, const lanelet::ConstLanelet & reference_lanelet,
+    lanelet::ConstLanelet * closest_lanelet, const double dist_threshold,
+    const double yaw_threshold) const;
```

Comment:
possible to return `std::optional<lanelet::ConstLanelet>` instead?

Reply:
I did that first, but changed it for sake of consistency